### PR TITLE
[RW-5997][risk=no]  Workspace cards should be open-able in a new tab

### PR DESCRIPTION
**Pull request by NehaBroad**
Screen Shot 2022-02-16 at 10 41 14 PM

https://user-images.githubusercontent.com/34481816/158066996-68ea318f-8563-4433-a2f8-f2d6b67c02e6.mov

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 1]
```diff
@@ -271,16 +292,18 @@ export const WorkspaceCard = fp.flow(withNavigation)(
                           )
                         }
                       >
-                        <div
+                        <StyledRouterLink
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-card.tsx:295]: This is why some of the tests are failing - because this is here, the workspace card test component needs to be wrapped in a `<MemoryRouter>`

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 2]
```diff
@@ -153,17 +155,24 @@ export const WorkspaceCard = fp.flow(withNavigation)(
       ]);
     }
 
-    onClick() {
+    trackWorkspaceNavigation() {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-card.tsx:158]: please add:
* a comment describing what this does 
* `boolean` return value
* description of what the return value means

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 3]
```diff
@@ -173,6 +182,13 @@ export const WorkspaceCard = fp.flow(withNavigation)(
       }
     }
 
+    workspaceCardNameTarget() {
+      const { tierAccessDisabled, workspace } = this.props;
+      return tierAccessDisabled
+        ? ' #'
+        : '/workspaces/' + workspace.namespace + '/' + workspace.id + '/data';
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-card.tsx:189]: nit: would be nicer with backticks, like `/workspaces/${namespace}` etc

Comment (NehaBroad) [reply ui/src/app/pages/workspace/workspace-card.tsx]: DOne

File: ui/src/app/pages/workspace/workspace-card.tsx (hunk 2 above)
Comment (NehaBroad) [reply ui/src/app/pages/workspace/workspace-card.tsx]: Hopefully done

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 4]
```diff
@@ -153,17 +155,29 @@ export const WorkspaceCard = fp.flow(withNavigation)(
       ]);
     }
 
-    onClick() {
+    // If enableResearchReviewPrompt is true and workspace needs Review:
+    // show the research purpose review modal and return false, to stop further navigation
+    // Else:
+    // Tracks the Analytics Event, Navigate to Featured workspace or Navigate to User workspace
+    // and return true to continue navigate to workspace detail page
+    trackWorkspaceNavigation(): boolean {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-card.tsx:163]: looks like the logic here is now exactly the same as it was before.  Is that correct?  The previous version is less complex than this one, so I'd prefer to revert the change.

Comment (NehaBroad) [reply ui/src/app/pages/workspace/workspace-card.tsx:163]: How is that less complex? This is just separating the navigation vs tracking, Also i need to work on it a little more as per our conversation on friday

Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-card.tsx]: Sorry!  I forgot that this was in-progress... LMK when you're ready for review or if you want to chat more.

**Comment by jmthibault79** [comment]
New version is #6474
